### DISCUSSION TITLE
Arguments at root query level in the schema?

### DISCUSSION THREAD

**Discussion by ms185730** [graphql-java/graphql-java#2399] [General]
Can you have have arguments in the root schema object in your schema definition? My Schema is as follows

schema {
  query(page: int, size: int): Query
}

type Query {
  books: [Book]!
  bookAndRelatedBooks(id: ID): BookAndRelatedBooks!
  findByName(name: String): [Book]!
}
...
Every query is paginated so I'd like to pass in arguments for pagination to the root query object, instead of having those arguments on each field in Query

The error is with the GraphQL Schema: 
`Failed to instantiate [graphql.GraphQL]: Factory method 'graphQL' threw exception; nested exception is SchemaProblem{errors=[InvalidSyntaxError{ message=Invalid Syntax ,locations=[SourceLocation{line=2, column=7}]}]}`

**Comment by bbakerman** [comment]
No you cant

That SDL

```
schema {
   query(page: int, size: int): Query
}
```

is not a field - its telling the schema the name of the root query type

eg

```
schema {
   query : MyWhackyNamedRootQueryType
}

type MyWhackyNamedRootQueryType {
   thisIsATopFieldWithArgs (arg : String) : Int
}
```

https://spec.graphql.org/June2018/#SchemaDefinition

**Reply by ms185730** [reply]
Gotcha, so my options are 
A) put my fields `books, bookAndRelatedBooks, findbyName` into another type and have the arguments on that type
or 
B) have the arguments on each field `books...`

**Reply by bbakerman** [reply]
right

many people use a namespace field to contain commons args or just generally break up their schema

```
type Query {
   accounting(commonArgs : String )  :  AccountingQuery 
   
  marketing(othercommonArgs : String )  :  MarketingQuery 
  }

```

**Reply by bbakerman** [reply]
```
query {
   accounting(....) {
        listAccounts

```
